### PR TITLE
Support for parsing the full API definition directly from the URL of the base resource

### PR DESCRIPTION
**Pull request by chefArchitect**
Added support for directly parsing a full API definition by using the
URL of the base resource - used a facade delegate function so that
backwards compatibility with the existing interface doesn't break.

Clients can either pass a URL string or an array of entities to the
function and depending on the type of the 1st argument, the delegate
calls the appropriate implementation.

Used the current implementation @ http://chefarchitect.github.io/swagger.ed/ and had to do the resource fetching manually.  I transferred my implementation in the project and now works with a single line of code

> SwaggerConverter.convert( 'base resource URL', function( err, api ){  });

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,23 +24,49 @@
...
+ * Definition of a 'convert()' facade function.
+ */
+var facade = function( ) {
+    
+  /*
+	 * check the type of the first argument if it's a string the return the
+	 * 'convertFromUrl' function - else return the default one
+	 */
+  if ( arguments.length === 0 )
+    throw "The convert() method expects at least 1 argument.";
```
Comment (mohsen1) [inline index.js]: Please throw an Error object.

File: index.js [hunk 2]
```diff
@@ -24,23 +24,49 @@
...
 /*
- * Converts Swagger 1.2 specs file to Swagger 2.0 specs.
- * @param resourceListing {object} - root Swagger 1.2 document where it has a
- *  list of all paths
- * @param apiDeclarations {array} - a list of all resources listed in
- * resourceListing. Array of objects
- * @returns {object} - Fully converted Swagger 2.0 document
-*/
+ * Converts Swagger 1.2 specs file to Swagger 2.0 specs. @param resourceListing
+ * {object} - root Swagger 1.2 document where it has a list of all paths @param
```
Comment (mohsen1) [inline index.js]: Please add each param and returns statement in one line

File: index.js [hunk 3]
```diff
@@ -102,11 +128,155 @@ function convert(resourceListing, apiDeclarations) {
...
+	}
+}
+
+
 /*
- * Builds "info" section of Swagger 2.0 document
- * @param source {object} - Swagger 1.2 document object
- * @returns {object} - "info" section of Swagger 2.0 document
-*/
+ * Builds "info" section of Swagger 2.0 document @param source {object} -
```
Comment (mohsen1) [inline index.js]: There are some JSDoc issues here too

File: test/test.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ var inputPath = './test/input/';
 var outputPath = './test/output/';
 
 require('mocha-jshint')();
-require('mocha-jscs')();
+//require('mocha-jscs')();
```
Comment (mohsen1) [inline test/test.js]: Please enable JSCS

File: index.js (hunk 1 above)
Comment (chefArchitect) [inline index.js]: ok

**Comment by mohsen1** [comment]
Thank yo for this. There are some code-style issues that I wish we could fix before merging. 
I'm testing the new functionality and will comment here if I found anything that needs to change.
You've added two new dependencies which is a little concern for browser version.

**Comment by chefArchitect** [comment]
Hey Mohsen, you're welcome.  Listen, before anything else please proceed with the merge only if you think that this functionality is useful regarding your initial goals.  As mentioned, I included your module in a tool I'm currently working on and found the lack of direct parsing a bit cumbersome. 

The cost of the extra dependencies comes to balance the convenience of not having to do the crawling manually.

**Comment by mohsen1** [comment]
Okay, so please hold on to it for now. 
We have [Swagger Tools](https://github.com/apigee-127/swagger-tools) that uses this module to convert Swagger files. Swagger Tools does have a CLI and a lot more dependencies. It makes sense to add this functionality to Swagger Tools `convert`  method and keep this module lean.

**Comment by chefArchitect** [comment]
Ok, fair enough.  Thanks for the feedback mate!

**Comment by prabhatjha** [comment]
@whitlockjc Can we adapt some of @chefArchitect work in swagger-tools ? Obviously it helped him solve his use case.

**Pull request closed without merging**